webapi/src/main/java/org/sakaiproject/webapi/controllers/TasksController.java

```diff
@@ -137,38 +132,24 @@ public Map<String, Object> getSiteTasks(@PathVariable String siteId) throws User
 
         data.put("canAddTask", taskService.canCurrentUserAddTask(siteId));
 
-        return data;
-    }
-    
-    @GetMapping(value = "/sites/{siteId}/users/current/isSiteUpdater", produces = MediaType.APPLICATION_JSON_VALUE)
-    public boolean isInstructorUser(@PathVariable String siteId) {
-        Session session = checkSakaiSession();
-
         try {
-            Site site = siteService.getSite(siteId);
             // Returns a boolean value which depends if an user is an instructor or not
-            return securityService.unlock(SiteService.SECURE_UPDATE_SITE, site.getReference());
+            if (securityService.unlock(SiteService.SECURE_UPDATE_SITE, site.getReference())) {
+                data.put("canUpdateSite", true);
+
+                Collection<Group> groups = site.getGroups();
+                if (!groups.isEmpty()) {
+                    Map<String, String> groupsMap = (Map<String, String>) groups.stream().collect(Collectors.toMap(Group::getId, Group::getTitle));
+                    data.put("groups", groupsMap.entrySet().toArray());
```

Comment:
Why do we need the map? If it needs to be an array, could it be something like `stream().map(Group::getTitle).toArray(String[]::new)`? Or could we rather collect the titles to a Set?

Reply:
Good spot! The whole groups json was broken. I've fixed the tasks tests and made sure it works in production.